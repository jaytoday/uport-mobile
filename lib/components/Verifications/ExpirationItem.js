// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
// Frameworks
import React from 'react'
import { Text, View, StyleSheet } from 'react-native'
import dateChecker from 'uPortMobile/lib/utilities/dateChecker'

// Styles
import { colors, fontLight } from 'uPortMobile/lib/styles/globalStyles'

const styles = StyleSheet.create({
  expirationDate: {
    fontSize: 12,
    lineHeight: 16,
    color: colors.grey74,
    fontFamily: fontLight,
  },
})
// Helpers
const ExpirationItem = props => {
  const formattedDate = dateChecker(props.d)
  return (
    <View>
      {!!props.d ? (
        <Text style={styles.expirationDate}>Exp: {formattedDate}</Text>
      ) : (
        <Text style={styles.expirationDate}>Exp: No Expiration</Text>
      )}
    </View>
  )
}

export default ExpirationItem
