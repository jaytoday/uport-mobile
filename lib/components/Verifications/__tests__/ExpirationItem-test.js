// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
/* globals it, expect */
import 'react-native'
import React from 'react'
import ExpirationItem from '../ExpirationItem'
import renderer from 'react-test-renderer'

describe('ExpirationItem', () => {
  it('renders ExpirationItem in seconds', () => {
    const date = 1583020800
    const tree = renderer.create(<ExpirationItem d={date} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders ExpirationItem in milliseconds', () => {
    const date = 1583020800000
    const tree = renderer.create(<ExpirationItem d={date} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
