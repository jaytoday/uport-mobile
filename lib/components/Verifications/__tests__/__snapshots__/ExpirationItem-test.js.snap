// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpirationItem renders ExpirationItem in milliseconds 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "rgba(74,74,74,1)",
        "fontFamily": undefined,
        "fontSize": 12,
        "lineHeight": 16,
      }
    }
  >
    Exp: 
    Feb 29th 2020
  </Text>
</View>
`;

exports[`ExpirationItem renders ExpirationItem in seconds 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "rgba(74,74,74,1)",
        "fontFamily": undefined,
        "fontSize": 12,
        "lineHeight": 16,
      }
    }
  >
    Exp: 
    Feb 29th 2020
  </Text>
</View>
`;
